synapse/storage/events.py

```diff
@@ -1908,9 +1908,9 @@ def _purge_history_txn(
             (room_id,)
         )
         rows = txn.fetchall()
-        max_depth = max(row[0] for row in rows)
+        max_depth = max(row[1] for row in rows)
 
-        if max_depth <= token.topological:
+        if max_depth < token.topological:
```

Comment:
I don't think we want to remove the equality check here.

Reply:
it does, because otherwise the tests fail :(